NXP-25935: Focus first field with error, and scroll to it

atoom75 opened the pull request:
Not sure at all with this code, but it seems to work. I make a lot of shadowRoot querySelector and it's seems not very generic and reliable. Any suggestion would be welcome!

Gabez0r commented on elements/document/nuxeo-document-edit.html:
```diff
@@ -91,7 +91,23 @@
+          return result;
+        } else {
+          var scrollableWrapper = this.$$('.scrollable');
+          var invalidField = this.$$('#document-edit').$.layout.$$('nuxeo-file-edit-layout').$$('[invalid]');
```
Our layouts are dynamically loaded according to the document's type. This will only for the edit layout of the `File` doctype.

Gabez0r replied:
`this.$['document-edit'].$.layout._getValidatableElements` returns you the list of validatable fields elements, regardless of the doctype.

Gabez0r commented on elements/document/nuxeo-document-edit.html:
```diff
@@ -91,7 +91,23 @@
+          if (invalidField !== undefined) {
+            var invalidFieldScrollTop = invalidField.scrollTop;
+            if (scrollableWrapper !== undefined) {
+              scrollableWrapper.scrollTo(0,invalidFieldScrollTop);
```
Assuming `field` is your validatable field element that is invalid, you can just do `field.scrollIntoView()`

atoom75 commented:
@Gabez0r I've tried to follow your suggestions with this commit: https://github.com/nuxeo/nuxeo-web-ui/pull/338/commits/3a29727f916111f7c8b3ea280a70b8a6bc6ac383
Thanks for your help!

Gabez0r commented:
👍

Gabez0r commented:
After [NXP-26010](https://jira.nuxeo.com/browse/NXP-26010), this logic needs to be moved to `nuxeo-document-form-layout.html`.

atoom75 commented:
@Gabez0r Ok I have moved and updated the validation code to `nuxeo-document-form-layout.html` and rewrited the git history. It seems to work as before the migration.

Gabez0r commented:
👍

nmpcunha commented:
👍
